pages/model/[uuid].tsx

```diff
@@ -111,17 +116,60 @@ function Model() {
   }
 
   useEffect(() => {
-    if (tab !== undefined && isTabOption(tab)) {
-      setGroup(tab)
+    if (currentUser && version) {
+      setIsManager(currentUser.id === version.metadata.contacts.manager)
+      setIsReviewer(currentUser.id === version.metadata.contacts.reviewer)
     }
-  }, [tab])
+  }, [currentUser, version])
+
+  const updateLastViewed = useCallback(
+    (role: string) => {
+      if (isManager && version?.updatedAt) {
+        const versionLastUpdatedAt = new Date(version?.updatedAt)
+        if (
+          (managerLastViewed && new Date(managerLastViewed).getTime() < versionLastUpdatedAt!.getTime()) ||
+          (reviewerLastViewed && new Date(reviewerLastViewed).getTime() < versionLastUpdatedAt!.getTime())
+        ) {
+          setShowLastViewedWarning(true)
+        }
+        putEndpoint(`/api/v1/version/${version?._id}/lastViewed/${role}`)
+      }
+    },
+    [managerLastViewed, reviewerLastViewed, version?._id, version?.updatedAt, isManager]
```

Comment:
Could replace the following dependencies `version?._id, version?.updatedAt` with simply `version`

Reply:
As far as I've tested, if you replace with just "version" it will continue to call because version is constantly updating.

Reply:
You're entirely right, go ahead with your solution